### ISSUE TITLE
Using without mono

### ISSUE THREAD

**Issue by Legogris** [OmniSharp/omnisharp-vscode#3557]
Is it possible to use the C# vscode extension without mono at all? On nixos the mono packaga erarely gets updated and it's a pain to have the extension broken with every other update. I'd wager most users of this extension have 0 use for mono.

**Comment by filipw** [comment]
yes OmniSharp ships with its own bundled Mono runtime so global Mono installation is not required

**Comment by Legogris** [comment]
> yes OmniSharp ships with its own bundled Mono runtime so global Mono installation is not required

Omnisharps bundled mono runtime does not work on NixOS (this is an inherent thing that can not be worked around). Guix, other architectures and various other setups will also run Core fine but can't run mono.I am asking about running core without mono

**Comment by filipw** [comment]
oh I thought the question was if global Mono was mandatory.

If by "Core" you mean ".NET Core" then the answer is currently no, OmniSharp is by design built on top of Mono to provide better/full coverage of the wide variety of MSBuild tasks and projects (.NET Core, Unity, desktop .NET and so on). This would not be possible if it was a .NET Core application since it wouldn't be able to i.e. run arbitrary code that could be part of MSBuild tasks. In fact, it was a .NET Core application at some point in the past but we moved away from it.

For NixOS I know that some people used to do some unofficial stuff like building Mono specifically there to work around the fact the original would be built on a differently configured system etc. There was also this package but it's out of date too https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/omnisharp-roslyn/default.nix I can't help here much since we really currently support OmniSharp on GuixSD/Nix like systems.

**Comment by Legogris** [comment]
Ah, it sounds kind of weird to me that you wouldn't be able to do any of those things from .NET Core (you can still have the entrypoint in .NET Core and run other binaries for .NET Framework when required, no?), but I guess that's just blissful ignorance on my side!

Take this as a feature request to drop the mono requirement, then I guess!

**Comment by filipw** [comment]
I will move this to omnisharp-roslyn repo.
